include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -303,6 +304,10 @@ class fm_index
                                    | views::join(delimiter)
                                    | views::to<std::vector<uint8_t>>;
 
+        // we need at least one delimiter
+        if (std::ranges::distance(text) == 1)
```

Comment:
just always add the delimiter makes the code a little easier I guess. And it shouldn't hurt.

Reply:
Somewhere, something breaks 🤷‍♂